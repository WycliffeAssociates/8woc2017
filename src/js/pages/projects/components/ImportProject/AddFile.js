import React from 'react';
import styled from 'styled-components';

export default class AddFile extends React.Component {

  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    event.preventDefault();
    if (this.fileInput) this.props.importProject(this.fileInput.files[0]);
    setTimeout(() => this.props.history.push('/progress'), 500);
  }

  componentDidMount() {
    const {txt} = this.props;
    var inputs = document.querySelectorAll('.inputFile');
    var submit = document.querySelector('.submit');
    if (submit) {
      submit.style.opacity=0;
    }
    Array.prototype.forEach.call (inputs, function(input) {
      var label = input.nextElementSibling,
        labelVal = label.innerHTML;

      input.addEventListener('change', function(e) {
        var fileName = '';
        fileName = e.target.value.split('\\').pop();

        if (fileName && labelVal!== txt.chooseProject)
        {
          label.querySelector('strong').innerHTML =fileName;
          submit.style.opacity=1;
        }
        else {
          label.innerHTML = labelVal;
          label.style.background= 'linear-gradient(to top, #0076FF, #00C5FF  )';
          submit.style.opacity=0;
        }

      });
    });
  }



  render() {
    const {txt} = this.props;
    return (
      <Form onSubmit={this.handleSubmit}>
        <Container>
          <Input type="file" name="file" id="file" className="inputFile" accept= ".zip"  innerRef={input => {
            this.fileInput = input;}} />
          <label htmlFor="file" style={styles.label}> <i className="material-icons">create_new_folder</i> <strong>{txt.chooseProject}</strong></label>
          <Submit className="submit" type="submit" > <strong>{txt.import}</strong> </Submit>
        </Container>
      </Form>
    );
  }

}
const styles = {
  label: {
    cursor: 'pointer',
    width: '55%',
    height: 'auto',
    color: 'white',
    background: 'linear-gradient(to top, #0076FF, #00C5FF  )',
    border: 'none',
    borderRadius: '5px',
    verticalAlign: 'middle',
    minHeight: '40px',
    maxHeight: '80px',
    minWidth: '190px',
    maxWidth: '250px',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    fontSize: '14px',
  },
};

const Form = styled.form`
  width: 100%;
  height: auto;
  padding: 5px;
  display: flex;
  align-items: center;
  justify-content: center;
`;
Form.displayName= 'Form';

const Container = styled.div`
  width: inherit;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  margin: auto;
`;
Container.displayName= 'Container';

const Submit = styled.button`
  background: green;
  margin-top: 10px;
  width: 55%;
  height: auto;
  color: white;
  border: none;
  border-radius: 5px;
  text-decoration: underline;
  i{
    vertical-align: middle;
  }
  min-height: 40px;
  min-height: 40px;
  max-height: 80px;
  min-width: 190px;
  max-width: 250px;
  cursor: pointer;
  font-size: 14px;
`;
Submit.displayName= 'Submit';
const Input = styled.input`
  border: none;
  width: 0.1px;
  height: 0.1px;
  opacity: 0;
  overflow: hidden;
  position: absolute;
  z-index: -1;
`;
Input.displayName= 'Input';
