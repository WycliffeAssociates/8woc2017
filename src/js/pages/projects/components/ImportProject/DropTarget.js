import React from 'react';
import styled from 'styled-components';
import AddFile from './AddFile';

export default class DropTarget extends React.Component {

  constructor(props) {
    super(props);
  }

  render() {
    //TODO add ability to drop files for upload
    const {txt} = this.props;
    return (
      <Container>
        <DashBorder>
          <h2> {txt.projectImport} </h2>

          <AddFile {...this.props} />
        </DashBorder>
      </Container>
    );
  }

}

const Container = styled.div`
  background: white;
  height: 50%;
  width: 35%;
  border-radius: 10px;
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 150px;
  min-width: 350px;

`;

const DashBorder = styled.div`
  border: 3px dashed lightgray;
  height: 97%;
  width: 97%;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  flex-direction: column;
  border-radius: 10px;
  padding: 10%;
  font-weight: 100;

`;
