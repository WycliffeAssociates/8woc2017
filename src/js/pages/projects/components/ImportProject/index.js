import React from 'react';
import styled from 'styled-components';
import img from '../../../../../assets/images/ProjectBG.svg';
import DropTarget from './DropTarget';

export default class ImportProject extends React.Component {

  constructor(props) {
    super(props);
  }


  render() {

    return (
      <Container>
        <h2 style={{fontWeight: '100'}}> There are no available projects </h2>
        <DropTarget {...this.props} />
      </Container>
    );
  }

}

const Container = styled.div`
  height: 100vh;
  width: 100vw;
  padding: 5%;
  position: absolute;
  top: 0;
  left: 0;
  background-image: url(${img}), linear-gradient(to bottom right, #F3F3F3 , #F3F3F3);
  background-size: 350px;
  background-repeat: repeat;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  flex-direction: column;
  font-weight: 200;

`;
