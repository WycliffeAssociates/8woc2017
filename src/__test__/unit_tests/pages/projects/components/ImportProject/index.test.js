/* global describe it expect */
import React from 'react';
import ImportProject from '../../../../../../js/pages/projects/components/ImportProject/';
import renderer from 'react-test-renderer';

describe('ImportProject test suite', () => {
  it('should render the component without exploding', () =>{
    const tree =
    renderer
      .create(<ImportProject />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
