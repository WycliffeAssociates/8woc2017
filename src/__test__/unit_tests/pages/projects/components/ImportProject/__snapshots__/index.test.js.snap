// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportProject test suite should render the component without exploding 1`] = `
<div
  className="sc-bZQynM iOFjkh"
>
  <h2
    style={
      Object {
        "fontWeight": "100",
      }
    }
  >
     There are no available projects 
  </h2>
  <div
    className="sc-ifAKCX gAured"
  >
    <div
      className="sc-EHOje ljiIeS"
    >
      <h2>
         Import a Project 
      </h2>
      <h3>
         Or 
      </h3>
      <form
        className="sc-bdVaJa cDjLwG"
        onSubmit={[Function]}
      >
        <div
          className="sc-bwzfXH jYANKA"
        >
          <input
            accept=".zip"
            className="inputFile sc-bxivhb jDxcbG"
            id="file"
            name="file"
            type="file"
          />
          <label
            htmlFor="file"
            style={
              Object {
                "alignItems": "center",
                "background": "linear-gradient(to top, #0076FF, #00C5FF  )",
                "border": "none",
                "borderRadius": "5px",
                "color": "white",
                "cursor": "pointer",
                "display": "flex",
                "fontSize": "14px",
                "height": "auto",
                "justifyContent": "center",
                "maxHeight": "80px",
                "maxWidth": "250px",
                "minHeight": "40px",
                "minWidth": "190px",
                "verticalAlign": "middle",
                "width": "55%",
              }
            }
          >
             
            <i
              className="material-icons"
            >
              create_new_folder
            </i>
             
            <strong>
              Choose Project...
            </strong>
          </label>
          <button
            className="submit sc-htpNat iuNMEV"
            type="submit"
          >
             
            <strong>
              Import
            </strong>
             
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
